`tests/lib/devicetree/legacy_api/libraries.devicetree.legacy` fails to build on pinnacle_100_dvk

When doing:

`/scripts/sanitycheck -s tests/lib/devicetree/legacy_api/libraries.devicetree.legacy -p pinnacle_100_dvk`

We get:

```
[  8%] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
In file included from /home/galak/git/zephyr/include/devicetree.h:31,
                 from /home/galak/git/zephyr/include/arch/arm/aarch32/arch.h:20,
                 from /home/galak/git/zephyr/include/arch/cpu.h:19,
                 from /home/galak/git/zephyr/include/kernel_includes.h:38,
                 from /home/galak/git/zephyr/include/kernel.h:17,
                 from /home/galak/git/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from /home/galak/git/zephyr/arch/arm/core/offsets/offsets.c:12:
/home/galak/git/zephyr/sanity-out/pinnacle_100_dvk/tests/lib/devicetree/legacy_api/libraries.devicetree.legacy/zephyr/include/generated/devicetree_legacy_unfixed.h:2332: error: "DT_FIXED_PARTITIONS_0_SIZE" redefined [-Werror]
 2332 | #define DT_FIXED_PARTITIONS_0_SIZE                  917504
      | 
In file included from /home/galak/git/zephyr/include/devicetree.h:31,
                 from /home/galak/git/zephyr/include/arch/arm/aarch32/arch.h:20,
                 from /home/galak/git/zephyr/include/arch/cpu.h:19,
                 from /home/galak/git/zephyr/include/kernel_includes.h:38,
                 from /home/galak/git/zephyr/include/kernel.h:17,
                 from /home/galak/git/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from /home/galak/git/zephyr/arch/arm/core/offsets/offsets.c:12:
/home/galak/git/zephyr/sanity-out/pinnacle_100_dvk/tests/lib/devicetree/legacy_api/libraries.devicetree.legacy/zephyr/include/generated/devicetree_legacy_unfixed.h:1950: note: this is the location of the previous definition
 1950 | #define DT_FIXED_PARTITIONS_0_SIZE                  98304
      | 
In file included from /home/galak/git/zephyr/include/devicetree.h:31,
                 from /home/galak/git/zephyr/include/arch/arm/aarch32/arch.h:20,
                 from /home/galak/git/zephyr/include/arch/cpu.h:19,
                 from /home/galak/git/zephyr/include/kernel_includes.h:38,
                 from /home/galak/git/zephyr/include/kernel.h:17,
                 from /home/galak/git/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from /home/galak/git/zephyr/arch/arm/core/offsets/offsets.c:12:
/home/galak/git/zephyr/sanity-out/pinnacle_100_dvk/tests/lib/devicetree/legacy_api/libraries.devicetree.legacy/zephyr/include/generated/devicetree_legacy_unfixed.h:2334: error: "DT_FIXED_PARTITIONS_0_LABEL" redefined [-Werror]
 2334 | #define DT_FIXED_PARTITIONS_0_LABEL                 "image-1"
      | 
In file included from /home/galak/git/zephyr/include/devicetree.h:31,
                 from /home/galak/git/zephyr/include/arch/arm/aarch32/arch.h:20,
                 from /home/galak/git/zephyr/include/arch/cpu.h:19,
                 from /home/galak/git/zephyr/include/kernel_includes.h:38,
                 from /home/galak/git/zephyr/include/kernel.h:17,
                 from /home/galak/git/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from /home/galak/git/zephyr/arch/arm/core/offsets/offsets.c:12:
/home/galak/git/zephyr/sanity-out/pinnacle_100_dvk/tests/lib/devicetree/legacy_api/libraries.devicetree.legacy/zephyr/include/generated/devicetree_legacy_unfixed.h:1952: note: this is the location of the previous definition
 1952 | #define DT_FIXED_PARTITIONS_0_LABEL                 "mcuboot"
      | 
cc1: all warnings being treated as errors
gmake[2]: *** [zephyr/CMakeFiles/offsets.dir/build.make:83: zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj] Error 1
```

--- a/tests/lib/devicetree/legacy_api/testcase.yaml
+++ b/tests/lib/devicetree/legacy_api/testcase.yaml
@@ -5,3 +5,4 @@ tests:
     # will mostly likely be the fastest.
     integration_platforms:
       - native_posix
+    platform_exclude:  pinnacle_100_dvk

